### PR TITLE
Write cookies to eTLD+1

### LINKED ISSUE #21377: Save amp-anaytics cookies in the highestAvailableDomain
We would like to save the cookies created by amp-analytics in the second level domain.
I see that such a functionality was added in the PR:

https://github.com/ampproject/amphtml/pull/1459/files

using the option: highestAvailableDomain

But it is not available in the amp-analytics tag.

Can it be added?

### PR DESCRIPTION
**Pull request by lannka**
Rewrote some of the tests to cover the cookie domain and expiration.

Closes #21377

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-cookie-writer.js [hunk 1]
```diff
@@ -236,106 +237,93 @@ describes.realWin('amp-analytics.cookie-writer', {
...
-      });
-      const cookieWriter = new CookieWriter(win, element, config);
-      sandbox.stub(cookieWriter.urlReplacementService_,
-          'expandStringAsync').callsFake(string => {
-        return Promise.resolve(string);});
-      return cookieWriter.write().then(() => {
-        expect(setCookieSpy).to.be.calledOnce;
-        expect(setCookieSpy).to.be.calledWith('testId', 'QUERY_PARAM(abc)');
-      });
+describes.fakeWin('amp-analytics.cookie-writer value', {amp: true}, env => {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-cookie-writer.js]: 👍 Way more cleaner after the rewrite.

File: extensions/amp-analytics/0.1/cookie-writer.js [hunk 2]
```diff
@@ -190,7 +190,9 @@ export class CookieWriter {
           // provide a way to overwrite or erase existing cookie
           if (value) {
             const expireDate = Date.now() + BASE_CID_MAX_AGE_MILLIS;
-            setCookie(this.win_, cookieName, value, expireDate);
+            setCookie(this.win_, cookieName, value, expireDate, {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/cookie-writer.js]: One question about the `highestAvailableDomain`. Would the old stored cookie be erased after the change?

Comment (lannka) [reply extensions/amp-analytics/0.1/cookie-writer.js]: kind of. you will be able to read both values. It's up to the reader which value to pick.
```
> document.cookie = 'abc=1; domain=example.com'
"abc=1; domain=example.com"
> document.cookie = 'abc=2; domain=www.example.com'
"abc=2; domain=www.example.com"
> document.cookie
"abc=1; abc=2"
```

Comment (zhouyx) [reply extensions/amp-analytics/0.1/cookie-writer.js]: I see, Thank you. Right now AMP doesn't provide a way to read cookies that are set in this way. But if we do in the future I believe we need to make sure to return the right cookie value. In that case what do you think we can do?

Comment (lannka) [reply extensions/amp-analytics/0.1/cookie-writer.js]: a safe bet is to read the last value (I think it's ordered by freshness).

Comment (zhouyx) [reply extensions/amp-analytics/0.1/cookie-writer.js]: Sounds good.

**Pull request merged**